pallets/gear-messenger/src/migrations.rs

```diff
@@ -0,0 +1,327 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Config, DispatchStash, Dispatches, Pallet, Waitlist};
+use common::storage::{Interval, LinkedNode};
+use frame_support::{
+    traits::{Get, GetStorageVersion, OnRuntimeUpgrade},
+    weights::Weight,
+};
+use gear_core::{
+    ids::MessageId,
+    message::{ContextStore, StoredDispatch},
+};
+use sp_std::marker::PhantomData;
+#[cfg(feature = "try-runtime")]
+use {
+    frame_support::codec::{Decode, Encode},
+    sp_std::vec::Vec,
+};
+
+pub struct MigrateToV3<T: Config>(PhantomData<T>);
+
+impl<T: Config> MigrateToV3<T> {
```

Comment:
It's not necessary to be impl of MigrateToV3. Btw just Into within the module seems ok